app/components/calendar-input.js

```diff
@@ -0,0 +1,74 @@
+import Ember from 'ember';
+
+export default Ember.TextField.extend({
+  tagName: 'input',
+  classNames: 'pickadate',
+  attributeBindings: [ "name", "type", "value", "id", 'required', 'pattern', 'placeholder', 'disableWeekends' ],
+
+  _getValidDate: function(selectedDate){
+    var today = new Date();
+    var currentDate = new Date();
+    var selected = selectedDate;
+    currentDate.setHours(0,0,0,0);
+    selected.setHours(0,0,0,0);
+    return currentDate > selected ? today : selectedDate;
+  },
+
+  isDateEqual: function(date) {
+    var selected = this.get('selection');
+    selected.setHours(0,0,0,0);
+    date.setHours(0,0,0,0);
+    return selected.getTime() === date.getTime();
+  },
+
+  didInsertElement() {
+    var _this = this;
+    var setting = false;
+
+    Ember.run.scheduleOnce('afterRender', this, function(){
+      Ember.$('.pickadate').pickadate({
+        format: 'ddd mmm d',
+        monthsFull: moment.months(),
+        monthsShort: moment.monthsShort(),
+        weekdaysShort: moment.weekdaysShort(),
+        disable: _this.get('disableWeekends') === true ? [1,2] : [],
+        clear: false,
+        today: false,
+        close: false,
+        min: moment().toDate(),
+
+        onClose: function() {
```

Comment:
Too many lines in `disInsertElement` callback. Would it be better to separate this function out in some named function outside `didInsertElement` callback?

Reply:
@Darshan-Chauhan I thought about it, but given that there seems to be a mix of scopes (`this` and `_this`) I figured it'd be clearer to leave it as it is.

Do you have any ideas ?
The callbacks need to be bound to the pickadate reference, but also need access to the ember component reference. I could start playing with binds, but I feel that makes it more complicated.

Reply:
You can use `call/apply/bind` for binding `pickadate` reference and pass `_this` as argument to function as component reference. 
I think use of `call` would be better to bind `pickadate` reference.

Reply:
To make it clearer it might be better to pass both of them as arguments I guess
Something like:

```javascript
// separate
function myOnClose(pickadate, component) { 
   // do things
}

// component
{
  onClose: function () {
    // this is bound to pickadate
   // _this is the component
    myOnClose.call(null, this, _this); 
  }
}
```

Reply:
No use of using call if we want to pass first argument as null, might as well use simple function and pass both this as an argument. 

```
// separate
function myOnClose(component) { 
    // here this will be available as pickadate reference.
   // do things
}

// component
{
  onClose: function () {
    // this is bound to pickadate
   // _this is the component
    myOnClose.call(this, _this);  //this bounds pickadate and _this gets passed as an argument
  }
}
```

This should work I think

Reply:
If there are too many arguments to pass then using `apply` would be better option.